### ISSUE TITLE
Doctrine DBAL 2.13 $statement->execute() returns bool (instead of Result)

### ISSUE THREAD

**Issue by knallcharge** [doctrine/dbal#4569]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.13.0

#### Summary

The release notes for DBAL 2.13 (https://www.doctrine-project.org/2021/03/29/dbal-2.13.html) stated that "_DBAL 3.0 extracts all fetch-methods from the Statement API and moved them to a new Result API that is returned from Statement::execute. **We have backported this API to 2.13_**"
It also gives an example that with 2.13 you could/should use code like this:
```
$result = $statement->execute();
while (($row = $result->fetchAssociative()) !== false) {
}
```
#### Current behaviour

$statement->execute() still returns only a boolean value instead of a resultset.

#### How to reproduce

- Upgrade to 2.13.0
- try to use fetch() on the result of $statement->execute()

#### Expected behaviour

$statement->execute() should return a resultset instead of bool.

**Comment by michaljusiega** [comment]
I can confirm, that changes in 2.13.0 is a little weird comparing to Changelog.

**Comment by beberlei** [comment]
Yes it looks we made a mistake there, will try to figure it out.

**Comment by michaljusiega** [comment]
> Yes it looks we made a mistake there, will try to figure it out.

Thank you.

**Comment by beberlei** [comment]
Fixed in #4580

**Comment by knallcharge** [comment]
Thanks for the fix. Side note: The release doc (https://www.doctrine-project.org/2021/03/29/dbal-2.13.html) still states that "$result = $statement->execute();" is to be used now. Updating to 2.13.1 reveals that execute() is now (too) deprecated, so the notes could need an update :)
